coderd/provisionerjobs_test.go

```diff
@@ -79,6 +80,16 @@ func TestProvisionerJobs(t *testing.T) {
 			job2, err := templateAdminClient.OrganizationProvisionerJob(ctx, owner.OrganizationID, job.ID)
 			require.NoError(t, err)
 			require.Equal(t, job.ID, job2.ID)
+
+			// Verify that job metadata is correct.
+			assert.Equal(t, job2.Metadata, &codersdk.ProvisionerJobMetadata{
+				TemplateVersionName: version.Name,
+				TemplateID:          template.ID,
+				TemplateName:        template.Name,
+				TemplateDisplayName: template.DisplayName,
+				WorkspaceID:         &w.ID,
+				WorkspaceName:       w.Name,
+			})
```

Comment:
I think it might be useful to assert the provisioner job metadata:
- on template import
- on template version create
- on workspace build

Reply:
I added a case for template import, I don't know of any distinction for "version create" so I skipped that.